### DISCUSSION TITLE
Concise lambda/anonymous function syntax in Mojo

### DISCUSSION THREAD

**Discussion by bomzj** [modular/modular#424] [Ideas]
Python has too verbose **lambda** keyword to define anonymous functions.
```python
lambda x : x + 1
lambda x, y : x + 1
```

I would like to see something more concise, yet expressive in Mojo e.g.
```fsharp
x -> x + 1
x y -> x + y
(x: int) y -> x + y
```

**Comment by lattner** [comment]
We're very likely to do this (probably following the scala3 approach) despite being generally against syntactic sugar but this isn't on the short term todo list given other priorities.

**Reply by frhack** [reply]
Python doesn't support multiline lambdas, I hope mojo does
